src/main/flight/pid.c

```diff
@@ -522,9 +526,11 @@ void pidController(const pidProfile_t *pidProfile, const rollAndPitchTrims_t *an
             gyroRateFiltered = dtermLowpassApplyFn((filter_t *) &dtermLowpass[axis], gyroRateFiltered);
             gyroRateFiltered = dtermLowpass2ApplyFn((filter_t *) &dtermLowpass2[axis], gyroRateFiltered);
 
-            const float rD = dynCd * MIN(getRcDeflectionAbs(axis) * relaxFactor, 1.0f) * currentPidSetpoint - gyroRateFiltered;    // cr - y
+            // no transition if relaxFactor == 0
+            const float transition = constrainf(getRcDeflectionAbs(axis) * relaxFactor, relaxFactor == 0, 1);
```

Comment:
Wouldn't it make more sense to put the 0 check on the outside:

```
float transition;
if (relaxFactor == 0) {
   transition = getRcDeflectionAbs(axis);
} else {
  transition = MIN(getRcDeflectionAbs(axis) * relaxFactor, 1.0f);
}
```

Reply:
It was done in a similar way before, but I opted for no explicit branching and uniform handling of both cases.

When relax is 0, constrain always returns 1, having no effect on setpoint.

Reply:
Using `bool` as `float` in this way is terrible.

Reply:
Why? Would you be more happy with `relax > 0 ? 0.f : 1.f`?

Reply:
if (relaxFactor == 0) {
   transition = 1;
} else {
  transition = MIN(getRcDeflectionAbs(axis) * relaxFactor, 1.0f);
}

Reply:
Okay, I would still like to know why it's terrible, having predicates converted to numbers allows to streamline and speed-up code in certain cases, so I don't agree that it's "terrible" in general.

Reply:
@DieHertz: `relaxFactor == 0` is a special case, and handled differently in both versions, and I think making this explicit increases readability.

Reply:
Partly it's from my C feeling that 'true' is anything nonzero.
Transition value 1.0f is partially a coincidence, it's hard to understand 
`(relaxFactor == 0) * 1.0f` may be a border case ...